frontend/src/views/UserView.vue

```diff
@@ -9,12 +9,30 @@
     </div>
     <p class="profile-view__bio">{{ bio }}</p>
     <div class="profile-view__username">{{ username }}</div>
-    <button
-      class="profile-view__button"
-      @click="($event) => $store.commit('connect')"
+    <div v-if="$store.state.connection.connectionStatus === 'pending'">
```

Comment:
extract this to computed/method. Also, do the null check before accessing connectionStatus